src/v/cloud_topics/l1/metastore.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+
+#include "base/seastarx.h"
+#include "cloud_topics/types.h"
+#include "container/fragmented_vector.h"
+#include "model/fundamental.h"
+#include "model/timestamp.h"
+#include "model/topic_revision.h"
+
+#include <seastar/core/future.hh>
+
+#include <expected>
+
+namespace experimental::cloud_topics::l1 {
+
+// Interface to interact with the L1 metastore. Meant to be totally agnostic to
+// the underlying implementation, whether it's an in-memory store, replicated
+// store, or even an external service.
+
+class metastore {
```

Comment:
Consider briefly describing the metastore, or linking to the design document.

Reply:
Done, added a high level overview. LMK if there's anything else you think would be helpful to include